### PR TITLE
Add INTEL_AUDIO_HAL selection flag default setting.

### PR DESCRIPTION
**Pull request by sguiriec**
@plbossart, @kalyankondapally please review this update for Audio HAL selection.

In order to be able to switch between baseline and Paramerter
Framework Audio HAL the flag INTEL_AUDIO_HAL needs to be propagated to
Baseline HAL.

Jira: None

Test: Build on latest manaifest of the 2 different Audio stacks.

Signed-off-by: Sebastien Guiriec <sebastien.guiriec@intel.com>

### CODE REVIEW

File: groups/audio/android_ia/BoardConfig.mk [hunk 1]
```diff
@@ -1,3 +1,7 @@
 BOARD_USES_ALSA_AUDIO := true
```
Comment (plbossart) [inline groups/audio/android_ia/BoardConfig.mk]: should we move this to false?

File: groups/audio/android_ia/BoardConfig.mk [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
 BOARD_USES_TINY_ALSA_AUDIO := true
 BOARD_USES_GENERIC_AUDIO ?= false
+# Audio HAL selvection Flag default setting.
```
Comment (plbossart) [inline groups/audio/android_ia/BoardConfig.mk]: typo

File: groups/audio/android_ia/BoardConfig.mk (hunk 1 above)
Comment (sguiriec) [inline groups/audio/android_ia/BoardConfig.mk]: Need to cross check with rest of the code. But Changing this flag should be part of different request.

**Comment by sguiriec** [comment]
Branch updated with fix

**Pull request closed without merging**